decrypt/decrypt.go

```diff
@@ -24,26 +22,15 @@ func File(path, format string) (cleartext []byte, err error) {
 	if err != nil {
 		return nil, fmt.Errorf("Failed to read %q: %v", path, err)
 	}
-	return Data(encryptedData, format)
+
+	store := common.DefaultStoreForPathOrFormat(path, format)
+	return DataWithStore(encryptedData, store)
 }
 
-// Data is a helper that takes encrypted data and a format string,
+// DataWithStore is a helper that takes encrypted data, and a store,
 // decrypts the data and returns its cleartext in an []byte.
-// The format string can be `json`, `yaml`, `dotenv` or `binary`.
-// If the format string is empty, binary format is assumed.
-func Data(data []byte, format string) (cleartext []byte, err error) {
-	// Initialize a Sops JSON store
-	var store sops.Store
-	switch format {
-	case "json":
-		store = &sopsjson.Store{}
-	case "yaml":
-		store = &sopsyaml.Store{}
-	case "dotenv":
-		store = &sopsdotenv.Store{}
-	default:
-		store = &sopsjson.BinaryStore{}
-	}
+func DataWithStore(data []byte, store sops.Store) (cleartext []byte, err error) {
```

Comment:
Do you have any use case for which this should be public?

Reply:
Choosing the store is the contentious part. There are several strategies:
- based on a format
- based on an extension
- based on whatever custom logic
Example: you can use `go-bindata` to include assets.

if `decrypt` has the only stable api, then basically that says "trust me i know what i am doing"

Reply:
Do you think `DataWithFormat` would be a more useful API? Considering `Store` is not really meant to be used by third parties. And I think we should re-export `Format` in this file as well. Keep in mind only the `decrypt` code should be public to external projects, and right now we're requiring them to access `go.mozilla.org/sops`.